android-libs/src/main/cpp/geoiplookup.cpp

```diff
@@ -2,65 +2,88 @@
 // Measurement-kit is free software. See AUTHORS and LICENSE for more
 // information on the copying conditions.
 
-#include "mkall.h"
+#include "io_ooni_mk_MKGeoIPLookupResults.h"
+#include "io_ooni_mk_MKGeoIPLookupTask.h"
 
 #include <limits.h>
 
-#include <measurement_kit/internal/geoiplookup/geoiplookup.h>
+#include <measurement_kit/internal/vendor/mkgeoip.hpp>
 
 #include "mkall_util.h"
 
 MKALL_GET_BOOLEAN(MKGeoIPLookupResults_Good,
-                  mk_geoiplookup_response_good,
-                  mk_geoiplookup_response_t)
+                  [](auto r) noexcept { return r->good; },
+                  mk::geoip::LookupResults)
 
 MKALL_GET_STRING(MKGeoIPLookupResults_GetProbeIP,
-                 mk_geoiplookup_response_ip,
-                 mk_geoiplookup_response_t)
+                 [](auto r) noexcept { return r->probe_ip.c_str(); },
+                 mk::geoip::LookupResults)
 
 MKALL_GET_STRING(MKGeoIPLookupResults_GetProbeASN,
-                 mk_geoiplookup_response_asn,
-                 mk_geoiplookup_response_t)
+                 [](auto r) noexcept { return r->probe_asn_string.c_str(); },
+                 mk::geoip::LookupResults)
 
 MKALL_GET_STRING(MKGeoIPLookupResults_GetProbeCC,
-                 mk_geoiplookup_response_cc,
-                 mk_geoiplookup_response_t)
+                 [](auto r) noexcept { return r->probe_cc.c_str(); },
+                 mk::geoip::LookupResults)
 
 MKALL_GET_STRING(MKGeoIPLookupResults_GetProbeOrg,
-                 mk_geoiplookup_response_org,
-                 mk_geoiplookup_response_t)
+                 [](auto r) noexcept { return r->probe_org.c_str(); },
+                 mk::geoip::LookupResults)
 
 MKALL_GET_LOGS(MKGeoIPLookupResults_GetLogs,
-               mk_geoiplookup_response_logs_size,
-               mk_geoiplookup_response_logs_at,
-               mk_geoiplookup_response_t)
+               [](auto r) noexcept { return r->logs.size(); },
+               [](auto r, size_t i) noexcept { return r->logs[i].c_str(); },
```

Comment:
This and other places could possibly return UTF-8 strings not compatible with MUTF-8. The proper thing to do would be to encoded to base64 if the string is not MUTF-8 compatible.

Reply:
Also, perhaps we can take advantage of the availability of a C++ string and use that directly.

Reply:
We should create an issue to do this later. This diff is too large.

Reply:
https://github.com/measurement-kit/android-libs/issues/80